### PR TITLE
Managed Beans 1.0 Specification

### PR DESCRIPTION
**Pull request by kwsutter**
Signed-off-by: Kevin Sutter <kwsutter@gmail.com>

## Specification PR template
When creating a specification project release review, create two PRs with the content divided as follows.

Include the following in PR#1:
- [x] A directory in the form wombat/x.y where x.y is the release major.minor version, and the directory contains the following.
- [x] Specification PDF in the form of wombat_x.y.pdf
- [x] Specification HTML in the form of wombat_x.y.html
- [x] A specification page named _index.md following the template at:
      https://github.com/jakartaee/specification-committee/blob/master/spec_page_template.md
- [x] For a Release Review, a summary that a Compatible Implementation
      is complete, passes the TCK, and that the TCK includes sufficient
      coverage of the specification. The TCK users guide MUST include
      the instructions to run the compatible implementations used to
      validate the release.  Instructions MAY be by reference.
- [x] The URL of the OSSRH staging repository for the api, javadoc:
      N/A - covered by [Jakarta Annotations](https://oss.sonatype.org/content/repositories/staging/jakarta/annotation/jakarta.annotation-api/1.3.5/)
- [x] The URL of the staging directory on downloads.eclipse.org for the proposed EFTL TCK binary:
      N/A - covered by [Platform TCK](http://download.eclipse.org/ee4j/jakartaee-tck/jakartaee8-eftl/promoted/eclipse-jakartaeetck-8.0.0.zip)
- [x] The URL of the compatibility certification request issue:
      [Glassfish Certification Request](https://github.com/eclipse-ee4j/jakartaee-platform/issues/99)
      [Managed Beans Certification Request](https://github.com/eclipse-ee4j/jakartaee-platform/issues/104)
These are basically the same since Glassfish is the Compatible Implementation for both.  Except, the latter one references the Managed Beans 1.0 Spec and is more consistent with the expected process.

### CODE REVIEW

**Comment by dblevins** [comment]
Spec PDF looks good.

File: managedbeans/1.0/_index.md [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+date: 2019-07-30
+summary: "Initial release of Managed Beans 1.0 for Jakarta EE 8"
+---
+Jakarta Managed Beans defines a set of basic services for container-managed objects
+with minimal requirements, otherwise known under the acronym “POJOs” (Plain Old Java Objects).
+
+* [Jakarta Managed Beans 1.0 Specification Document](./ManagedBeans-spec-1.0.pdf) (PDF)
+* [Jakarta Managed Beans 1.0 Specification Document](./ManagedBeans-spec-1.0.html) (HTML)
+* Jakarta Managed Beans 1.0 Javadoc - N/A (covered by Jakarta Annotations)
+* Jakarta Managed Beans 1.0 TCK - N/A  (covered by Glassfish TCK)
```
Comment (bshannon) [inline managedbeans/1.0/_index.md]: "Glassfish TCK" -> "Jakarta EE platform TCK"

**Comment by scottkurz** [comment]
In addition to bshannon's comments, (missing copyright and managedbeans/_index.md )....

1. The link to compatibility certification request issue in the PR comment, where it says _N/A - covered by Glassfish Certification Request_  links back to this same PR: `https://github.com/jakartaee/specifications/pull/19`    Not sure what agreements or procedures have been established for using a single Glassfish Certification Request, but looking at this for the first time, it seems like there maybe should still be a link to a certification issue, but one that mentions all the APIs it is covering.    
* Similar comment for the .../1.0/_index.md listing under **Compatible Implementations**
2. Within the spec, the "author" (I don't claim to be correctly stating the name of this field/heading/whatever) is listed as:
`Jakarta EE Platform Team, https://projects.eclipse.org/projects/ee4j.jakartaeeplatform`  
Should this refer to the `https://projects.eclipse.org/projects/ee4j.ca` project maybe?

File: managedbeans/1.0/_index.md [hunk 2]
```diff
@@ -0,0 +1,36 @@
+---
+title: "Jakarta Managed Beans Spec 1.0"
+date: 2019-07-30
+summary: "Initial release of Managed Beans 1.0 for Jakarta EE 8"
+---
+Jakarta Managed Beans defines a set of basic services for container-managed objects
+with minimal requirements, otherwise known under the acronym “POJOs” (Plain Old Java Objects).
```
Comment (scottkurz) [inline managedbeans/1.0/_index.md]: “POJOs”  => "POJOs"  (ascii 34 quote char)

File: managedbeans/1.0/_index.md (hunk 1 above)
Comment (kwsutter) [reply managedbeans/1.0/_index.md]: got it. thanks.

File: managedbeans/1.0/_index.md (hunk 2 above)
Comment (kwsutter) [reply managedbeans/1.0/_index.md]: Wow!  how did you find that?  :-)  Will clean up.

**Comment by kwsutter** [comment]
> In addition to bshannon's comments, (missing copyright and managedbeans/_index.md )....
> 
>     1. The link to compatibility certification request issue in the PR comment, where it says _N/A - covered by Glassfish Certification Request_  links back to this same PR: `https://github.com/jakartaee/specifications/pull/19`    Not sure what agreements or procedures have been established for using a single Glassfish Certification Request, but looking at this for the first time, it seems like there maybe should still be a link to a certification issue, but one that mentions all the APIs it is covering.
> 
> 
>     * Similar comment for the .../1.0/_index.md listing under **Compatible Implementations**
> 

Yeah, chicken and the egg type of situation...  There is no standalone TCK for Managed Beans, so it has to be covered by the Jakarta EE TCK test execution and the eventual Certification Request for Eclipse Glassfish.  Or, maybe this would be better covered by the Annotation TCK execution.  Since this is a Spec only release (no API delivered), maybe this item should be removed from the checklist?  Will have to discuss further...

> 
>     1. Within the spec, the "author" (I don't claim to be correctly stating the name of this field/heading/whatever) is listed as:
>        `Jakarta EE Platform Team, https://projects.eclipse.org/projects/ee4j.jakartaeeplatform`
>        Should this refer to the `https://projects.eclipse.org/projects/ee4j.ca` project maybe?
> 

No, this will stay with the Platform.  Eventually, the ManagedBeans spec will be contributed by Oracle and we'll have a real Specification.  Also, we've already been discussing that the Jakarta Annotations project should be merged with the Platform post Jakarta EE 8.

>     2. I'd kind of think this level of nitpicking wouldn't matter..but in case there's any tooling or something... I'll note there's a different case btw. the dir name `managedbeans` and the spec filename `ManagedBeans-spec-1.0.pdf`...

This is fair.  Since I have to make other updates, I'm going to get rid of the camel case on the generated file names.

**Comment by kwsutter** [comment]
>     * spec docs need Eclipse Foundation copyright
> 
>     * managedbeans/_index.md is missing

Yeah, my mistake on both fronts.  The changes were on my system, but I hadn't pushed them and re-generated the docs.  Will be done shortly.  Thanks.

**Comment by scottkurz** [comment]
Just to be clear @kwsutter ... I mentioned the project home only because I noticed that https://projects.eclipse.org/projects/ee4j.jakartaeeplatform  is a dead link.

**Comment by kwsutter** [comment]
> Just to be clear @kwsutter ... I mentioned the project home only because I noticed that https://projects.eclipse.org/projects/ee4j.jakartaeeplatform is a dead link.

Ahh...  this is just bad formatting on the front page of the pdf.  The actual link has the hyphen in it.  https://projects.eclipse.org/projects/ee4j.jakartaee-platform.  I think we're good.

**Comment by scottkurz** [comment]
> Ahh... this is just bad formatting on the front page of the pdf. The actual link has the hyphen in it. https://projects.eclipse.org/projects/ee4j.jakartaee-platform. I think we're good.

Well, sorry for the confusion, but I botched my comment.   Bear with me for this one:

1) The displayed text in the PDF and HTML is **correct**:  https://projects.eclipse.org/projects/ee4j.jakartaee-platform

2) The URL that you get if you actually click the link is **incorrect**:  it is wrongly: `https://projects.eclipse.org/projects/ee4j.jakartaee-platform1.0` 

Weirdly enough, the reason for my confusion is that I clicked the link first, saw it didn't work, assumed it was the same URL as the displayed text, so mouse-selected and copied-then-pasted that String.   Well... Windows 10 + Acrobat actually smashes together  the `endLine1-beginLine2` into `endLine1beginLine2` when you copy/paste this way.  Weird and too much information I know !

**Comment by scottkurz** [comment]
Nevermind... sorry for my confusion...the link appears to be rendered by the PDF client program, not the document itself.   It works as expected when I open this PDF in MS Edge and Chrome, not in Acrobat 2017 though.   

This all starting innocently enough with me just clicking the link :)  Learned more than I'd wanted to know here.

File: managedbeans/_index.md [hunk 3]
```diff
@@ -0,0 +1,4 @@
+---
+title: "Jakarta Managed Beans"
+summary: "Jakarta Managed Beans defines a set of basic services for container-managed objects with minimal requirements, otherwise known under the acronym "POJOs" (Plain Old Java Objects)."
```
Comment (bshannon) [inline managedbeans/_index.md]: The nested double quotes are probably going to be a problem.
Perhaps convert the inner quotes to single quotes?
Or figure out how to escape them with backslash?
Sorry, I have no idea what hugo requires.

File: managedbeans/1.0/_index.md [hunk 4]
```diff
@@ -0,0 +1,36 @@
+---
+title: "Jakarta Managed Beans Spec 1.0"
+date: 2019-07-30
+summary: "Initial release of Managed Beans 1.0 for Jakarta EE 8"
+---
+Jakarta Managed Beans defines a set of basic services for container-managed objects
+with minimal requirements, otherwise known under the acronym "POJOs" (Plain Old Java Objects).
+
+* [Jakarta Managed Beans 1.0 Specification Document](./ManagedBeans-spec-1.0.pdf) (PDF)
+* [Jakarta Managed Beans 1.0 Specification Document](./ManagedBeans-spec-1.0.html) (HTML)
```
Comment (bshannon) [inline managedbeans/1.0/_index.md]: The file names changed to lowercase below but not in these links.

File: managedbeans/_index.md (hunk 3 above)
Comment (kwsutter) [reply managedbeans/_index.md]: I did some searching and I couldn't figure out how hugo was going to respond either.  So, I'm taking the safe way out and just removed the embedded quotes.  The acronym POJO is common enough that it doesn't have to be in quotes.  I changed it here and in the release _index.md just to be consistent.  Thanks.

File: managedbeans/1.0/_index.md (hunk 4 above)
Comment (kwsutter) [reply managedbeans/1.0/_index.md]: Thanks, Bill!  Done.

**Comment by starksm64** [comment]
Need links to platform certification request and TCK results

**Comment by starksm64** [comment]
# Spec Review Checklist

1. Spec PR
  - [x] PR uses [template](https://github.com/jakartaee/specifications/blob/master/pull_request_template.md)
  - [x] Directory of form {spec}/x.y
  - [x] PDF of form {spec}-spec-x.y.pdf ("-spec" preferred but not required)
  - [x] HTML of form {spec}-spec-x.y.html ("-spec" preferred but not required)
  - [x] Index page {spec}/x.y/_index.md following [template](https://github.com/jakartaee/specification-committee/blob/master/spec_page_template.md)
  - [x] Index page {spec}/_index.md following [template](https://github.com/jakartaee/specification-committee/blob/master/spec_index_template.md)
  - [x] No other files (e.g., no jakarta_ee_logo_schooner_color_stacked_default.png)
  - [x] Staging repository link of the form https://oss.sonatype.org/conte/repositories/staging/jakarta/{spec}/jakarta-{spec}-api/x.y.z/
  - [x] EFTL TCK link of the form http://download.eclipse.org/.../+.zip
  - [x] Compatibility certification link of the form https://github.com/eclipse-ee4j/{project}/#{issue}
  - [x] Second PR for just apidocs (NA)

2. _index.md
  - [x] Link to spec pdf
  - [x] Link to spec html
  - [x] Link to apidocs
  - [x] Link to final TCK download zip file of the form https://download.eclipse.org/jakartaee/{spec}/x.y/*{spec}-tck-x.y.z.zip
  - [x] Link to API jar file of the form https://search.maven.org/artifact/jakarta.{spec}/jakarta.{spec}-api/x.y.z/jar
  - [x] Name of and link to at least one Compatible Implementation

3. javadocs
  - [ ] Footer contains Eclipse copyright and link to license
  - [ ] ESFL license is included, usually as doc-files/speclicense.html
  - [ ] no META-INF directory in PR

4. Spec PDF
  - [ ] Correct spec title
  - [ ] Version number of the form x.y, not x.y.z
  - [ ] Correct Eclipse copyright line
  - [ ] No DRAFT or SNAPSHOT

5. Spec HTML
  - [ ] Same as PDF

6. TCK zip file
  - [ ] README file (optional for this release)
  - [ ] EFTL license file, preferably named LICENSE.md
  - [ ] User's Guide (or equivalent documentation)
  - [ ] How to test the Compatible Implementation(s) listed in _index.md above with the TCK (may be in UG)

7. TCK User's Guide (or equivalent documentation)
  - [ ] Software requirements listed
  - [ ] Installation and configuration described
  - [ ] How to run tests
  - [ ] Where to file challenges

8. Compatibility certification request
  - [ ] Request follows [template](https://github.com/jakartaee/specification-committee/blob/master/compatibility-certification-request.md)
  - [ ] SHA-256 fingerprint matches staged TCK zip file
  - [ ] Request issue has `certification` label.

9. TCK results summary
  - [ ] Page is hosted by Compatible Implementation project
  - [ ] Includes all information from certification request
  - [ ] Summary includes number of tests passed, failed, errors

**Comment by kwsutter** [comment]
> Need links to platform certification request and TCK results
> https://eclipse-ee4j.github.io/glassfish/certifications/jakarta-full-profile/8.0/TCK-Results
> cert request TBD

Should be ready to go now.  Other than the Certification Request, is the TCK Results URL supposed to be posted somewhere?  I couldn't find any other spot to put that URL.

**Comment by starksm64** [comment]
> 
> Should be ready to go now. Other than the Certification Request, is the TCK Results URL supposed to be posted somewhere? I couldn't find any other spot to put that URL.
It goes in the cert compatibility request.

**Comment by bshannon** [comment]
The cert request has been filed: https://github.com/eclipse-ee4j/jakartaee-platform/issues/99

**Comment by bshannon** [comment]
I believe we decided this is ready to go, but needs to synchronize with the platform.
I'll remove the draft label.

**Comment by bshannon** [comment]
@kwsutter This PR should refer to the compatibility certification request in [issue #104](https://github.com/eclipse-ee4j/jakartaee-platform/issues/104).

**Comment by kwsutter** [comment]
> @kwsutter This PR should refer to the compatibility certification request in [issue #104](https://github.com/eclipse-ee4j/jakartaee-platform/issues/104).

Is that we decided?  I didn't see a reply to my last note on this topic, so I didn't know if we had made a decision.

**Comment by kwsutter** [comment]
> > @kwsutter This PR should refer to the compatibility certification request in [issue #104](https://github.com/eclipse-ee4j/jakartaee-platform/issues/104).
> 
> Is that what we decided? I didn't see a reply to my last note on this topic, so I didn't know if we had made a decision.

I just added the #104 CR to the Description.

**Comment by dblevins** [comment]
@kwsutter mark ready for review when you get the chance.  (saw you caught the management api pr already)

**Comment by kwsutter** [comment]
@dblevins Added the checklist for you...  Just trying to get ahead of the game on this final day of approvals...  :-)

* On ballot completion
 - [x] The designated specification committee representative adds this final checklist to the main PR.
 - [ ] The specification committee representative merges the specification and apidocs PRs.
 - [ ] The designated specification committee representative calculates the staged EFTL TCK signature and promotes it to the committee download area
  using the https://ci.eclipse.org/jakartaee-spec-committee/job/promote-release/ job.
 - [ ] The specification project member who created the staging release promotes the specification api jars to maven central. An example release job script can be found here https://wiki.eclipse.org/MavenReleaseScript.
 - [ ] The EMO updates the specification page with the ballot results.
This list goes on the committed spec index page.
 - [ ] The specification project team should go through the merged spec website page to verify all the links are valid.
 - [ ] The specification project team should approve the compatibility request.
 - [ ] The compatible implementation project/vendor MUST send an email tck@eclipse.org for approval of the compatible implementation for trademark usage.
 - [ ] The specification project team should merge any final release branch as appropriate for the branch management for the project.

**Pull request merged**